src/services/item/plugins/invitation/repository.ts

```diff
@@ -84,16 +87,27 @@ export const InvitationRepository = AppDataSource.getRepository(Invitation).exte
       invitations
         .filter(
           (i) =>
-            !existingEntries.find(({ email, item }) => email === i.email && item.id === itemPath),
+            // exclude duplicate item-email combinations that are already invited
+            !existingEntries.find(({ email, item }) => email === i.email && item.path === itemPath),
         )
         .map((invitations) => ({
           ...invitations,
           item: { path: itemPath },
           creator,
         })),
     );
-    // TODO: optimize
-    return this.getMany(insertResult.identifiers.map(({ id }) => id));
+
+    const ids = insertResult.identifiers.map(({ id }) => id);
+    if (ids.length) {
+      // get the created invitations
+      const res = await this.createQueryBuilder('invitation')
+        .innerJoinAndSelect('invitation.item', 'item')
+        .innerJoinAndSelect('invitation.creator', 'creator')
+        .where('invitation.id IN (:...ids)', { ids })
+        .getMany();
+      return res;
+    }
```

Comment:
Is it necessary?

Reply:
I replaced it, in order to not get the result of version. Do you think it should return the result of ?